embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/api/EmbraceApiService.kt

```diff
@@ -60,27 +60,32 @@
         if (cachedResponse.isValid()) { // only bother if we have a useful response.
             request = request.copy(eTag = cachedResponse.eTag)
         }
-        val response = apiClient.executeGet(request)
-        return when (response.statusCode) {
-            HttpURLConnection.HTTP_OK -> {
+        return when (val response = apiClient.executeGet(request)) {
+            is ApiResponse.Success<String> -> {
                 logger.logInfo("Fetched new config successfully.")
                 val jsonReader = JsonReader(StringReader(response.body))
                 serializer.loadObject(jsonReader, RemoteConfig::class.java)
             }
-
-            HttpURLConnection.HTTP_NOT_MODIFIED -> {
-                logger.logInfo("Confirmed config has not been modified.")
-                cachedResponse.remoteConfig
-            }
-
-            ApiClient.NO_HTTP_RESPONSE -> {
-                logger.logInfo("Failed to fetch config (no response).")
-                null
+            is ApiResponse.Failure -> {
+                when (response.code) {
+                    HttpURLConnection.HTTP_NOT_MODIFIED -> {
```

Comment:
`NOT_MODIFIED` is 304 and shouldn't be considered a failure. This is another reason why expanding the number of sealed classes would be useful.

Reply:
Updated.